yt/funcs.py

```diff
@@ -1310,3 +1311,17 @@ def sglob(pattern):
     Return the results of a glob through the sorted() function.
     """
     return sorted(glob.glob(pattern))
+
+
+def dictWithFactory(factory: Callable[[Any], Any]) -> dict:
```

Comment:
you'll need to include `from typing import Type`
```suggestion
def dictWithFactory(factory: Callable[[Any], Any]) -> Type:
```

Also I'm not very happy about using `Any` here, especially given that the function is only used with very limited set of types (at least for now). I think `Callable` might suffice as an annotation here.
Also note that we'd need to define a custom type such as
`FieldType = Union[str, Tuple[str, str]]`, which would belong in `yt._typing` (#3064)

Reply:
Actually `Callable[[Any], Any]` is stricter than `Callable`, since it allows only *one* argument to be passed to the factory.